### ISSUE TITLE
NoMongo: ADMIN / USER REDESIGN: Event Venues

### ISSUE THREAD

**Issue by palisadoes** [PalisadoesFoundation/talawa-admin#1928]
**Is your feature request related to a problem? Please describe.**

We now have event venues in the API. We need to apply them properly to the Event modal.

**Describe the solution you'd like**

1. Make the Event modal select from a list of event venues, or `Null` which would appear in the UI/UX as blank to reduce confusion.
2. Create tests for this update. All tests must be valid and pass
3. Update the sample data with this new functionality
1. Use our official figma file for guidance. There may have been changes in the design.
    - https://docs.talawa.io/docs/design/ux-design-system

**Describe alternatives you've considered**

- N/A

**Approach to be followed (optional)**

- N/A

**Additional context**

- N/A

**Potential internship candidates**

Please read this if you are planning to apply for a Palisadoes Foundation internship 

- https://github.com/PalisadoesFoundation/talawa/issues/359

**Comment by varshith257** [comment]
I would like to work on this issue

**Comment by sahildkun** [comment]
I would like to work on this issue

**Comment by karthxk07** [comment]
@palisadoes Can I work on this?

**Comment by palisadoes** [comment]
Unassigning. Inactivity

**Comment by akshayw1** [comment]
@palisadoes Can I work on this issue?

**Comment by akshayw1** [comment]
Could you elaborate the issue,the below is the event modal, want exactly is needed?

**Comment by varshith257** [comment]
@akshayw1 Talawa API now had support for Event venues. We need to support the same here. Check talawa-api for more

**Comment by akshayw1** [comment]
@varshith257 Do we have any demo content that. would added in sample data?

**Comment by palisadoes** [comment]
unassigning, inactivity

**Comment by Ameerjafar** [comment]
@palisadoes can you assign me this issue

**Comment by Ameerjafar** [comment]
![Screenshot 2024-10-24 184656](https://github.com/user-attachments/assets/11fa175d-3e0f-4d12-b85e-22e919929351)
this is where should I add the venue or I have to add the venue in somewhere else @pranshugupta54 ?

**Comment by Ameerjafar** [comment]
@pranshugupta54 basically program will happen in multiple venues?

**Comment by palisadoes** [comment]
Unassigning inactivity

**Comment by VanshikaSabharwal** [comment]
@palisadoes  . Can you please assign me this issue if no one is working

**Comment by palisadoes** [comment]
Is this in the same style as other dropdowns in the app?

**Comment by palisadoes** [comment]
unassigning. Inactivity.

**Comment by VanshikaSabharwal** [comment]
Hey @palisadoes. Can you please reassign me this issue. I am working on this.

**Comment by VanshikaSabharwal** [comment]
Hello @palisadoes  @Dante291 I have to update sample data in talawa-api in order to have list of venues. So, do i have to make another issue for tht in talawa-api and then get it assigned or directly raise PR.

**Comment by palisadoes** [comment]
Yes, that would be helpful.

**Comment by VanshikaSabharwal** [comment]
I am working on this issue.

**Comment by VanshikaSabharwal** [comment]
@palisadoes  @pranshugupta54  I have updated the UI 

![Screenshot from 2025-01-05 11-56-21](https://github.com/user-attachments/assets/a1e40054-0939-4c10-86d6-f2efa918a124)
![Screenshot from 2025-01-05 11-46-27](https://github.com/user-attachments/assets/759c02e2-6fb9-451f-8d4c-7d0ae95493ba)

and added data in talawa-api but i am not able to retrieve it on frontend. the data is there in db . 

![Screenshot from 2025-01-05 11-58-59](https://github.com/user-attachments/assets/60ec9e65-2757-49fe-b232-8fe616c7ae02)

Can you please me figure out how can i get the data from db. 
Right now i have data in mongodb . 
![image](https://github.com/user-attachments/assets/ccea2a57-f20c-4acf-8d29-ad77fc06fead)

But i am not able to retrieve it maybe because of the query parameters
![image](https://github.com/user-attachments/assets/0a36d6c1-70ce-4dab-893f-3d3fefaa9834)
 
I am using this GET_VENUES query to get data but it gives undefined in frontend logs.
![image](https://github.com/user-attachments/assets/3ac9bb9c-f622-4ac8-82ff-a4474e57be7c)
![image](https://github.com/user-attachments/assets/2323edfc-1b2b-4f72-871f-9c1040a94f27)

**Comment by palisadoes** [comment]
1. Use igraphql as explained in the documentation to formulate the query 
1. Look at the API schema file to evaluate what queries are available

**Comment by VanshikaSabharwal** [comment]
Okay @palisadoes  i got it.

**Comment by VanshikaSabharwal** [comment]
![image](https://github.com/user-attachments/assets/4c5f68d1-1761-4de0-8d41-dd26c41e6c98)
I have implemented this functionality. @palisadoes @varshith257 
Can you please tell me is there anything else to do.

**Comment by palisadoes** [comment]
1. This meets the requirement
1. Make the screens you edit comply with our design guide. The exact screens may not be there but there will be many similar ones that you can use.
    - https://www.figma.com/design/dmKt00m9GVSeA1nebnkxql/Untitled?node-id=0-1&node-type=canvas&t=TPzfh1DE9IxjzsEe-0

**Comment by VanshikaSabharwal** [comment]
Yes @palisadoes I am doing it.

**Comment by palisadoes** [comment]
We created an this issue to do basic changes to the CSS colors for consistency

- https://github.com/PalisadoesFoundation/talawa-admin/issues/2880
- https://github.com/PalisadoesFoundation/talawa-admin/pull/2984

The resulting PR created a comprehensive design system for the entire app. This was unexpected. The approach was a long term goal that became immediately implementable. We had to act promptly.

We have just merged the PR. There will be a follow up PR to adjust some colors as the Figma design has lettering that is sometimes hard to read.

When this is done we'll need your assistance to merge the screens into the `develop` branch with this issue.

If your screens have not been affected, then you should use the new methodology to refactor them.

This is an extraordinary circumstance. I hope you will appreciate the decision.

**Comment by palisadoes** [comment]
Please proceed. The changes mentioned previously are mostly in components

**Comment by VanshikaSabharwal** [comment]
@palisadoes Thank you for the update. I completely understand the urgency and importance of acting promptly in this situation. I appreciate the effort to create a comprehensive design system for the entire app, and I’m happy to help with merging the screens into the develop branch once the follow-up PR is ready.

I will ensure that any screens of mine that haven’t been affected will be refactored using the new methodology. Let me know if there’s anything specific you need assistance with during this process.

**Comment by VanshikaSabharwal** [comment]
@palisadoes I want to ask, right now I am trying to store the venues along with other data in the database as part of the create venue functionality. Does this issue include the problem of adding venues to events, which requires updating the schema for mutations? When I tried to do this, it did not work. Could it be because we are trying to migrate to PostgreSQL?

**Comment by VanshikaSabharwal** [comment]
Because as mentioned in this issue. 
1.  Make the Event modal select from a list of event venues, or Null which would appear in the UI/UX as blank to reduce confusion.
  - I did it successfully.
   
2. Create tests for this update. All tests must be valid and pass
  - i am doing it right now and it will be completed soon once i get to know more about if there are any enhancements left. 
  
3. Update the sample data with this new functionality
   - Somehow there was already existing data which i used. So, there was actually no need to update data.

**Comment by palisadoes** [comment]
Use the component styling from this PR for the popups
- https://github.com/PalisadoesFoundation/talawa-admin/pull/2984

**Comment by VanshikaSabharwal** [comment]
![Image](https://github.com/user-attachments/assets/f28a734b-65c7-48ae-84d5-81f7261a885c)

![Image](https://github.com/user-attachments/assets/494a10ed-6aa7-4475-a99f-31238c5e1f76)

**Comment by VanshikaSabharwal** [comment]
@palisadoes i have updated the design to blue shades.

**Comment by palisadoes** [comment]
1. The event moral has already been designed. It was referenced previously here
    1. https://github.com/PalisadoesFoundation/talawa-admin/pull/2984#issuecomment-2585396521

**Comment by VanshikaSabharwal** [comment]
![Image](https://github.com/user-attachments/assets/ce5eebd4-9354-4e65-8989-a609d06d82f8)
Is there any reason for this calender covering only half of the section.

**Comment by palisadoes** [comment]
1. The event moral has already been designed. It was referenced previously here. You'll have to add the drop down
    1. https://github.com/PalisadoesFoundation/talawa-admin/pull/2984#issuecomment-2585396521

**Comment by palisadoes** [comment]
Merge with the latest upstream. The event calendar now matches Figma. Your design does not.

**Comment by VanshikaSabharwal** [comment]
Okay sir

**Comment by VanshikaSabharwal** [comment]
I upstreamed my branch and is it fine now? 
@palisadoes I also want to ask that is this looking better with before version or after version because it was green before so i changed it to blue shades. On upstream i did not find this blue shades on buttons thus i updated it. 
Before:
![Image](https://github.com/user-attachments/assets/b73ed693-780f-43e8-9b39-33b9c7ddcebb)
After:
![Image](https://github.com/user-attachments/assets/d799bcc5-5e9b-424d-b245-0eb6c21e14bf)

**Comment by palisadoes** [comment]
It still doesn't match Figma exactly:

![Image](https://github.com/user-attachments/assets/1b7e77f0-e103-4191-8c28-87cd3535a23d)

**Comment by palisadoes** [comment]
The today button you created is a good addition

**Comment by VanshikaSabharwal** [comment]
Yes Thank you @palisadoes I will improve the styling and reach out to you with updates

**Comment by VanshikaSabharwal** [comment]
@palisadoes I don't understand why even after upstreaming i i do not see this updated styling according to figma

**Comment by palisadoes** [comment]
1. The last CSS work on it was PR #2883
    1. https://github.com/PalisadoesFoundation/talawa-admin/blame/develop-postgres/src/components/EventCalendar/YearlyEventCalender.tsx
1. Please fix this so that the look and feel is restored.

**Comment by VanshikaSabharwal** [comment]
Yes @palisadoes

**Comment by palisadoes** [comment]
I just realized that there was a specific issue for fixing events. Please focus on the event venues instead. 
- https://github.com/PalisadoesFoundation/talawa-admin/issues/3333

Sorry for the confusion.

**Comment by palisadoes** [comment]
1. You have not used the modal merged in this PR for your design. The modal is the new standard it explains what to do very clearly in the source code. This is the third time that I've posted this.
    1. https://github.com/PalisadoesFoundation/talawa-admin/pull/2984#issuecomment-2585396521
1. Other
    ![Image](https://github.com/user-attachments/assets/2e6876d6-632c-44b8-8c70-0576210ed8d5)
1. Please remove the `Today` button. I forgot this capability is in one of the dropdowns. It's not aligned in an aesthetic way.
    ![Image](https://github.com/user-attachments/assets/9410ef8f-0d9c-450f-ada4-df79aa90163e)

**Comment by palisadoes** [comment]
1. Are you using the `develop-postgres` branch as the base of your changes?
2. This is what is there currently

![Image](https://github.com/user-attachments/assets/69e894db-25a8-4d43-b13d-0870fa60ed7a)

**Comment by VanshikaSabharwal** [comment]
> 1. Are you using the `develop-postgres` branch as the base of your changes?
> 2. This is what is there currently
> 
> ![Image](https://github.com/user-attachments/assets/69e894db-25a8-4d43-b13d-0870fa60ed7a)

Yes @palisadoes I am using develop-postgres branch as base

**Comment by VanshikaSabharwal** [comment]
should i try rebasing it with develop branch??

**Comment by palisadoes** [comment]
No

**Comment by VanshikaSabharwal** [comment]
@palisadoes Can you please tell me how can i import sample-data base as we are migrating to postgres.

**Comment by palisadoes** [comment]
Use the develop branch of the API until further notice.

**Comment by VanshikaSabharwal** [comment]
Okay @palisadoes

**Comment by VanshikaSabharwal** [comment]
@palisadoes I cloned the develop-postgres branch of talawa-admin and removed the today button as you suggested. 

Before :- 
![Image](https://github.com/user-attachments/assets/8b0fbab8-905f-48c1-a095-73d994b14e63)

After :- 

![Image](https://github.com/user-attachments/assets/3275a35a-5482-41b8-b3e1-975fab0fc800)

![Image](https://github.com/user-attachments/assets/748f9ebe-e292-48ea-ae3e-91548aab7e1d)

**Comment by palisadoes** [comment]
1. Please leave the button there. My suggestion was a mistake.
1. Why are the events in the last image not visible?
1. Please submit a PR when complete

**Comment by VanshikaSabharwal** [comment]
@palisadoes it's not visible because when i took the ss that time it was just trying to load.

**Comment by palisadoes** [comment]
Please submit a PR

**Comment by VanshikaSabharwal** [comment]
Yes I am submitting a PR

**Comment by VanshikaSabharwal** [comment]
@palisadoes I saw that there is no code in API to include venue as well as a part of new created event. First we will have to update the API for it and then only we can make the frontend work

**Comment by palisadoes** [comment]
Yes. The front end code will soon be compatible with the new API. Probably in the next few hours

**Comment by VanshikaSabharwal** [comment]
Okay So after that I can raise PR i guess

**Comment by VanshikaSabharwal** [comment]
Unassigning in order to work on other more important issues at this moment. Will get back to this issue soon. Once talawa api has good test code coverage for `develop-postgres` branch. Also first API must have venue being created as a part of create event. So we'll have to first work on API

**Comment by VanshikaSabharwal** [comment]
@palisadoes  What do you recommend me doing. I have updated code om which i worked for several month in order to resolve this issue. Should i wait for API to have functionality for venue list being added as a part of created events??

**Comment by palisadoes** [comment]
Create an issue in the API to create the functionality

**Comment by VanshikaSabharwal** [comment]
Yes I am doing it @palisadoes

**Comment by VanshikaSabharwal** [comment]
Please unassign me this issue

**Comment by gurramkarthiknetha** [comment]
I want to work on this issue .could you please assign this issue @palisadoes

**Comment by Cioppolo14** [comment]
Unassigning due to inactivity and/or no submitted PR. Others need to get a chance. Please let us know whether you want to be reassigned before someone else requests assignment or select another issue.

**Comment by gurramkarthiknetha** [comment]
i am working on it i will open pr soon .
please assign this issue @Cioppolo14

**Comment by mpdesai90** [comment]
Our policy is to assign no more than two issues to each contributor across all repositories. This way everyone gets a chance to participate in the projects. We sometimes give exceptions for more urgent cases and sometimes we lose track, but the policy stands. You have reached your limit, please wait until your existing issues are closed before requesting more issues. You could unassign yourself from one of the other issues too.

**Comment by bandhan-majumder** [comment]
@palisadoes is this feature for `develop` branch? For `develop-postgres`, I think it will be better to keep this issue on hold until the migration completes.

**Comment by VanshikaSabharwal** [comment]
Yes @palisadoes first backend must be completed for this issue which is this issue. https://github.com/PalisadoesFoundation/talawa-api/issues/2969  then only the frontend can be made as then only the PR tests will fail if backend .
 Also @palisadoes I made whole frontend for this issue but due to urgent issues which must be completed i had to leave this issue in between. Whoever is working on this issue can take a look at this frontend i made. https://github.com/PalisadoesFoundation/talawa-admin/issues/1928#issuecomment-2611511164

**Comment by VanshikaSabharwal** [comment]
@bandhan-majumder previously this issue was for `develop` branch.

**Comment by gurramkarthiknetha** [comment]
@VanshikaSabharwal i am working on the backend for this issue

**Comment by gurramkarthiknetha** [comment]
I have some queries. Could we connect on Slack to discuss them?

**Comment by gurramkarthiknetha** [comment]
i want to work on this i am also working on the backend of this @palisadoes

**Comment by mpdesai90** [comment]
Our policy is to assign no more than two issues to each contributor across all repositories. This way everyone gets a chance to participate in the projects. We sometimes give exceptions for more urgent cases and sometimes we lose track, but the policy stands. You have reached your limit, please wait until your existing issues are closed before requesting more issues. You could unassign yourself from one of the other issues too.

**Comment by gurramkarthiknetha** [comment]
I have a good idea on how to fix this issue. My two PRs are almost complete; only the code coverage part is left.

**Comment by gurramkarthiknetha** [comment]
unassign the https://github.com/PalisadoesFoundation/talawa-api/issues/2969 and assign this issue

**Comment by gurramkarthiknetha** [comment]
i will complete this first

**Comment by Cioppolo14** [comment]
@gurramkarthiknetha  You have 3 issues assigned to you right now.  I will not assign you another one until you back below 2 issues.

**Comment by AryanSwaroop** [comment]
can this issue be assigned to me ?

**Comment by palisadoes** [comment]
How's progress on this? It needs to be completed as a GSoC prerequisite.

**Comment by palisadoes** [comment]
Unassigning. 2 weeks of inactivity

**Comment by VanshikaSabharwal** [comment]
Hey @palisadoes I previously completed the frontend part but that time backend was not ready due to which i could not raise PR. If the backend for this is ready i can make frontend again??

**Comment by palisadoes** [comment]
Are you asking to be assigned?

**Comment by VanshikaSabharwal** [comment]
yeah @palisadoes i can take this issue if API work for this done. else first API work will have to be done. Because if backend is not ready then the tests will fail in PR. So, Is backend work done for this??

**Comment by palisadoes** [comment]
1. What issue is preventing you from doing this? 
2. Is it assigned and do you want to tackle it?
3. If you want to be assigned to the other issue, you'll need to comment in that issue

**Comment by VanshikaSabharwal** [comment]
@palisadoes  Actually, I am worried that if I take this issue now and the API is not ready, it will delay the issue, and I might not be able to successfully merge my PR because the tests will fail unless the API is available in talawa-api.

That's the only thing holding me back from requesting assignment for this issue.

Can anyone please confirm if the backend is ready for this issue? Previously, it was incomplete, which prevented me from merging the PR even after finishing the frontend work. I hope the same doesn't happen this time, so I want to ensure that the backend work for this feature is completed.

**Comment by MayankJha014** [comment]
@palisadoes  Assign this to me

**Comment by Cioppolo14** [comment]
@VanshikaSabharwal Are you talking about if this issue was completed: https://github.com/PalisadoesFoundation/talawa-api/issues/2969?  It does not appear to be. If @VanshikaSabharwal  does not want this,  @MayankJha014 Have you read all the discussion above and feel you can complete this?

**Comment by Adhish1612Acharya** [comment]
@palisadoes  I have gone through the discussion above . I would love the opportunity to work on it. Please let me know if I can proceed by assigning it to me.

**Comment by VanshikaSabharwal** [comment]
Yes @Cioppolo14  the api is not completed. I would suggest you @Adhish1612Acharya to first complete the api then start working on this frontend issue. I would be happy to assist you.

**Comment by Adhish1612Acharya** [comment]
@Cioppolo14 Got it! Can the API issue be assigned to me so I can work on it first

**Comment by palisadoes** [comment]
Unassigned. Inactivity

**Comment by vamshi567bh** [comment]
@palisadoes please assign me

**Comment by soumajitgh** [comment]
@palisadoes can i work on it?

**Comment by palisadoes** [comment]
Unassigning due to inactivity and/or no submitted PR. Others need to get a chance. Please let us know whether you want to be reassigned before someone else requests assignment or select another issue.

Also, our policy is to assign no more than two issues to each contributor across all repositories. This way everyone gets a chance to participate in the projects. We sometimes give exceptions for more urgent cases and sometimes we lose track, but the policy stands. You have reached your limit, please wait until your existing issues are closed before requesting more issues. You could unassign yourself from one of the other issues too.
